sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,523 @@
+```
+ 
+The common recorder provides the following public methods and properties:
+
+- `record`: Which deals with recording and playing back the network requests,
+  depending on the value assigned to the `TEST_MODE` environment variable. If
+  `TEST_MODE` equals to `record`, it will automatically store network requests
+  in a plain text file in the folder `recordings` at the root of your
+  repository (which for our example case is the root of the
+  `@azure/keyvault/keyvault-keys` repository).
+  This package assumes that the tests in the sdk are leveraging
+  [mocha](https://mochajs.org/) and [rollup](https://rollupjs.org/guide/en/)
+  (and [karma](https://karma-runner.github.io/latest/index.html) test runner
+  for browser tests) as suggested by the [template](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/template/template)
+  package in the repo. It also returns an object with a method `stop()`, which
+  will allow you to control when you want the recorder to stop re-routing your
+  http requests.
+- `env`, which exposes the environment variable's object from either NodeJS or
+  the browser (useful on isomorphic tests).
+- `delay`, which is an asynchronous function that will resolve once the given milliseconds have elapsed,
+  but only while the `TEST_MODE` is not `playback`, since you want to make sure you can run the playback tests
+  as fast as possible.
+- `isRecordMode`, which is a shorthand for checking if the environment variable
+  `TEST_MODE` is set to `record`.
+- `isPlaybackMode`, which is a shorthand for checking if the environment
+  variable `TEST_MODE` is set to `playback`.
+- `setReplaceableVariables`, which will allow you to hide sensitive content
+  from the environment variables (more on that later).
+- `setReplacements`, which will allow you to hide sensitive content by doing
+  pattern matching in the output files.
+- `skipQueryParams`, since query parameters may contain sensitive information,
+  the array provided to method will signal what query parameters to remove from
+  the recordings.
+
+### Configuring your project
+
+Having the common recorder as a devDependency means that you'll be able to start
+recording tests right away by using the exported method `record`. We'll get
+into the details further down this document. This function will do recordings,
+or will play back previous recordings, depending on an environment variable:
+`TEST_MODE`. If the environment variable `TEST_MODE` is empty, `record` (and most
+of the functions provided by test-utils-recorder) won't be doing anything. You'll need
+to set this environment variable to `record` to start recording, and then to
+`playback` to play the recordings back at your code.
+
+#### package.json scripts
+
+You can write scripts in your `package.json` to
+make it easier to switch from record mode to playback mode, on a meaningful context, as follows:
+
+    "integration-test:node": "mocha myNodeTests.js",
```

Comment:
Also, things that go into files should be in code blocks to show that they're meant to be treated separately